packages/features/settings/TimezoneChangeDialog.tsx

```diff
@@ -4,109 +4,122 @@ import { useSession } from "next-auth/react";
 import { useEffect, useState } from "react";
 
 import dayjs from "@calcom/dayjs";
+import type { ManipulateType as DayjsManipulateType } from "@calcom/dayjs";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { trpc } from "@calcom/trpc/react";
-import { Dialog, DialogClose, DialogContent, DialogFooter, showToast } from "@calcom/ui";
-
-const TimezoneChangeDialogContent = ({
-  onAction,
-  browserTimezone,
-}: {
-  browserTimezone: string;
-  onAction: (action?: "update" | "cancel") => void;
-}) => {
-  const { t } = useLocale();
+import { showToast, Dialog, DialogClose, DialogContent, DialogFooter, DialogHeader } from "@calcom/ui";
+
+function hideDialogFor(hideFor: [number, DayjsManipulateType], toastContent: string) {
+  document.cookie = `calcom-timezone-dialog=1;max-age=${
+    dayjs().add(hideFor[0], hideFor[1]).unix() - dayjs().unix()
+  }`;
+  if (toastContent) showToast(toastContent, "success");
+}
+
+const TimezoneChangeDialogContent = () => {
+  const { t, isLocaleReady } = useLocale();
+  const { open, setOpen, browserTimezone } = useOpenTimezoneDialog();
+
+  if (!isLocaleReady) return null;
+
   const utils = trpc.useUtils();
   const formattedCurrentTz = browserTimezone.replace("_", " ");
 
-  // save cookie to not show again
-  function onCancel(hideFor: [number, dayjs.ManipulateType], toast: boolean) {
-    onAction("cancel");
```

Comment:
Remove very dirty interface, also with onAction("update") - these onActions which used to dismiss the Dialog are now handled automatically by event bubbling from DialogClose.